frontend/src/components/UserUpdateSfdsAuth.vue

```diff
@@ -0,0 +1,283 @@
+        { text: 'Mailbox', value: 'm' },
+        { text: 'Uses', value: 'u' },
+        { text: 'Permission', value: 'p' },
+        { text: 'Actions', value: 'actions', sortable: false }
+      ],
+      sfdsAuthorizations: [],
+      sfdsAuthorizationsToSet: [],
+      sfdsMailboxes: ['hscis', 'bcma', 'phc', 'hoopc'],
+      sfdsUses: ['bcma', 'hscis', 'phc', 'wda', 'hoopc', 'grp'],
+      sfdsPermissions: ['get', 'send', 'get-send', 'get-delete', 'get-send-delete'],
+      currentSfdsAuthorization: {},
+      editDialog: false,
+      deleteDialog: false,
+      editedIndex: -1,
+      defaultAuthorization: {
+        m: '',
+        u: [],
+        p: ''
+      },
+      alertStatus: false,
+      alertType: "error",
+      alertMessage: ''
+    };
+  },
+  async created() {
+    let sfdsAuthString = "";
+
+    if (this.$store.state.user.attributes.sfds_auth_1) {
+      sfdsAuthString = sfdsAuthString.concat(this.$store.state.user.attributes.sfds_auth_1[0]);
+    }
+    if (this.$store.state.user.attributes.sfds_auth_2) {
+      sfdsAuthString = sfdsAuthString.concat(this.$store.state.user.attributes.sfds_auth_2[0]);
+    }
+    if (this.$store.state.user.attributes.sfds_auth_3) {
+      sfdsAuthString = sfdsAuthString.concat(this.$store.state.user.attributes.sfds_auth_3[0]);
+    }
+    if (this.$store.state.user.attributes.sfds_auth_4) {
+      sfdsAuthString = sfdsAuthString.concat(this.$store.state.user.attributes.sfds_auth_4[0]);
+    }
+
+    if (sfdsAuthString !== "") {
+      this.sfdsAuthorizations = JSON.parse(sfdsAuthString);
+    } else {
+      this.sfdsAuthorizations = [];
+    }
+  },
+  watch: {
+    editDialog (val) {
+      val || this.closeDialog()
+    },
+    deleteDialog (val) {
+      val || this.closeDialog()
+    }
+  },
+  computed: {
+    dialogTitle () {
+      return this.editedIndex === -1 ? 'New Authorization' : 'Edit Authorization'
+    },
+    mailboxRules() {
+      const rules = []
+      const rule1 = v => !!v || 'A mailbox is required'
+      rules.push(rule1)
+
+      if (this.sfdsAuthorizations.length>0) {
+        // Check all existing SFDS authorizations for unique mailbox
+        if (this.editedIndex === -1) {
+          const rule2 = v => !(this.sfdsAuthorizations.some( sfdsAuth => sfdsAuth['m'] === v)) || 'Mailbox must be unique';
+          rules.push(rule2);
+        // Check all existing SFDS authorizations except the currently selected one for unique mailbox
+        } else {
+          const authsToMatch = JSON.parse(JSON.stringify(this.sfdsAuthorizations));
+          authsToMatch.splice(this.editedIndex, 1);
+          const rule2 = v => !(authsToMatch.some( sfdsAuth => sfdsAuth['m'] === v)) || 'Mailbox must be unique';
+          rules.push(rule2);
+        }
+      }
+      return rules;
+    }
+  },
+  methods: {
+    editItem: function (item) {
+      this.editedIndex = this.sfdsAuthorizations.indexOf(item)
+      this.currentSfdsAuthorization = Object.assign({}, item)
```

Comment:
Weird method, `Object.assign`. Zero way to guess what's happening: just have to know what `item`'s properties are copied into `{}` and returned to be assigned to `this.currentSfdsAuthorization`. You could also have written `Object.assign(this.currentSfdsAuthorization, item)` to the same effect. Not a suggestion, just thinking through it.